src/linter/worker.go

```diff
@@ -106,9 +105,11 @@ func (w *Worker) ParseContents(filename string, contents []byte, lineRanges []gi
 	b := w.ctx.scratchBuf
 	b.Reset()
 	b.ReadFrom(rd)
-	contents = append(make([]byte, 0, b.Len()), b.Bytes()...)
 
-	waiter := BeforeParse(len(contents), filename)
+	contents := append(make([]byte, 0, b.Len()), b.Bytes()...)
+	file.ResetContents(contents)
```

Comment:
Is it possible to create a new `File` here instead of changing existing one?

I also don't think that it's a win to replace FileInfo with File during the filesystem traversal.
We sometimes don't have a file "contents", so our `File` can be in two states: with data and without date.
It would be simpler if `File` is guaranteed to always be a file with contents.
It's an entity used for linter, while FileInfo is mostly for the traversal-related code and to create`File` later.